drive: add 123Pan cloud storage backend

Higanoneko opened the pull request:
<!--
Thank you very much for contributing code or documentation to rclone! Please
fill out the following questions to make it easier for us to review your
changes.

You do not need to check all the boxes below all at once, feel free to take
your time and add more commits. If you're done and ready for review, please
check the last box.
-->

#### What is the purpose of this change?

Add a new backend for 123Pan (123 Cloud Storage), a Chinese cloud storage service.

**Features implemented:**
- Full file operations: list, upload, download, copy, move, delete
- Directory operations: mkdir, rmdir, purge
- MD5 hash support
- Instant upload - files that already exist on 123Pan servers are uploaded instantly
- Share link creation via `rclone link`
- OAuth-style authentication using client_id and client_secret from 123Pan Developer Console
- Automatic VIP level detection with adaptive rate limiting
- Proper filename character encoding for restricted characters

**Limitations (123Pan API limitation):**
- No modification time support
- Empty files (0 bytes) are not supported
- API rate limits vary by user VIP level

[123Pan Official API Documents](https://123yunpan.yuque.com/org-wiki-123yunpan-muaork/cr6ced)

Test passed:

<img width="1868" height="516" alt="PixPin_2025-12-16_19-51-18" src="https://github.com/user-attachments/assets/05d56e26-2344-4c7a-a53c-4c2477c2aa0c" />

[123pan test result files](https://github.com/user-attachments/files/24190699/123pan_test.zip)


#### Was the change discussed in an issue or in the forum before?

This is the refactored version of Pull Request #9017 

This will close #9015 

#### Checklist

- [x] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-new-feature-or-bug-fix).
- [x] I have added tests for all changes in this PR if appropriate.
- [x] I have added documentation for the changes if appropriate.
- [x] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [x] I'm done, this Pull Request is ready for review :-)

Higanoneko committed:
drive: add 123Pan cloud storage backend

Higanoneko committed:
docs: update README.md for 123pan

Higanoneko committed:
123pan: add purge/cleanup support and improve uploads

- Implement fs.Purger and fs.CleanUpper interfaces
- Add streaming upload for large files (>100MB)
- Update QPS limits per official documentation
- Refactor API retry logic with exponential backoff

Higanoneko committed:
123pan: disable some test for better daily test
disable bisync and gitannex for better test in daily task.

PianCat commented:
@ncw @roucc 
If there is anything I can help to improve at any time, please let me know, and I would be happy to provide assistance with improvements.

Higanoneko committed:
123pan: improve backend of Seek and Logs
fix Seek error due to AsyncReader not support
improve VIP detect and log outpus

Higanoneko committed:
123pan: update encoder config

roucc reviewed (approved):
Hi @PianCat sorry for the delay. We are ready to merge this once the CI goes green as it all looks good, thanks!

0oo000 commented:
我在使用 rclone copy 进行 123上传的时候出现内存溢出的情况，即使线程只有 1 个，文件数量比较多文件大小在 500MB～70GB

Higanoneko committed:
fix(123pan): 修复内存溢出问题，防止大文件/大目录 OOM

- 非 seekable 大文件上传改用磁盘临时文件代替内存 RepeatableReader
- 移除 listAllFiles/collectAllFileIDs/collectTrashedFileIDs 全量收集函数
- forEachFile 改为流式分页遍历，不再 accumulate 全部文件
- purgeCheck 改用 BFS 队列遍历 + 批量 flush，删除 deleteTree 递归收集
- CleanUp 改用 BFS 遍历 + 批量 flush，删除 collectTrashedFileIDs 递归收集
- uploadSlice 复用 Fs.sliceHTTPClient 避免每次创建 http.Client
- maxMemoryUploadSize 从 100MB 降至 32MB
- sliceBuffer 新增 clampSliceSize 上限 16MB

0oo000 commented:
大佬可以看一下这个bug，rclone使用openapi上传文件，当文件夹名中有特殊字符、符号时会使文件名发送改变，例如文件名“世界计划：无法歌唱的初音未来 (2025)”上传后名称会发送改变“世界计划‛：无法歌唱的初音未来 (2025)”，webdav协议下使用rclone lsd查看文件名可以正常显示，opaenapi协议下使用rclone lsd文件名显示异常（和123云盘文件夹内名称不一致），附件中123-webdav显示是正常的文件名，123-openapi中显示的文件名异常
[rclone-123.txt](https://github.com/user-attachments/files/28064275/rclone-123.txt)
